Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -1069,31 +1215,156 @@ public class ProgramBuilder {
         return perform(LoadRegExp(value: value, flags: flags)).output
     }
 
-    @discardableResult
-    public func createObject(with initialProperties: [String: Variable]) -> Variable {
-        // CreateObject expects sorted property names
-        var propertyNames = [String](), propertyValues = [Variable]()
-        for (k, v) in initialProperties.sorted(by: { $0.key < $1.key }) {
-            propertyNames.append(k)
-            propertyValues.append(v)
+    public struct Property {
```

Comment:
What about methods? Do we also want to support those?

Reply:
I suppose we can support methods natively. The existing approach to generate a function and then add it as an input to a property worked for us so I didn't think it necessary to add methods as a bunch of separate operations.

Having a rethink about this, adding methods would probably make type tracking easier (i.e. we can bundle this functionality in ObjUtils.swift and expose one API to get property and method information)? and we will now be able to use `super` within object method definitions. WDYT?

Reply:
On the flip side though, we'd have to generate object method operations for each function variant (i.e. PlainFunc, Async, Generator, etc). And at some point we'd also have to extend this to classes.

Reply:
Right. I think we could go with just a single `addMethod` for now, then see how that plays out, and then see if we still want to support other types of methods as well later on

Reply:
For now I've added a `addMethod`, `addComputedMethod` and `addGeneratorMethod` functions. The first two occur quite frequently in the JSC test repo and there's a few samples that use generator methods. At some point I'd like to refactor class creation so that we can reuse method and property creation code from object creation.